lib/Dialect/D2M/Transforms/GlobalDataFormatConversion.cpp

```diff
@@ -0,0 +1,137 @@
+// SPDX-FileCopyrightText: (c) 2025 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "ttmlir/Dialect/D2M/Transforms/Passes.h"
+#include "ttmlir/Dialect/TTCore/IR/TTCore.h"
+#include "ttmlir/Dialect/TTIR/Utils/Utils.h"
+
+#include "mlir/Transforms/DialectConversion.h"
+
+namespace mlir::tt::d2m {
+#define GEN_PASS_DEF_D2MGLOBALDATAFORMATCONVERSION
+#include "ttmlir/Dialect/D2M/Transforms/Passes.h.inc"
+
+namespace {
+
+static std::optional<ttcore::DataType>
+parseTargetFormat(llvm::StringRef format) {
+  if (format == "f32") {
+    return ttcore::DataType::Float32;
+  } else if (format == "bf16") {
```

Comment:
:warning: **llvm\-else\-after\-return** :warning:
do not use `` else `` after `` return ``
```suggestion
  } if (format == "bf16") {
```

Reply:
It should now be fixed!